sql/api/src/main/scala/org/apache/spark/sql/types/TimeType.scala

```diff
@@ -33,6 +34,10 @@ import org.apache.spark.sql.errors.DataTypeErrors
 @Unstable
 case class TimeType(precision: Int) extends AnyTimeType {
 
+  if (!SqlApiConfHelper.confGetter.get()().isTimeTypeEnabled) {
```

Comment:
It's tricky to perform validations in a constructor. We didn't do so for geo types either.